tartufo/scanner.py

```diff
@@ -871,64 +919,38 @@ def chunks(self) -> Generator[types.Chunk, None, None]:
             "Branches to be scanned: %s",
             ", ".join([str(branch) for branch in branches]),
         )
-
+        branch_cnt = 0
+        branch_len = len(branches)
         for branch_name in branches:
             self.logger.info("Scanning branch: %s", branch_name)
             if branch_name == "HEAD":
                 commits = [self._repo.get(self._repo.head.target)]
             else:
                 branch = self._repo.branches.get(branch_name)
                 try:
-                    commits = self._repo.walk(
-                        branch.resolve().target, pygit2.GIT_SORT_TOPOLOGICAL
+                    commits = list(
+                        self._repo.walk(
+                            branch.resolve().target, pygit2.GIT_SORT_TOPOLOGICAL
+                        )
```

Comment:
I'm guessing `self._repo.walk()` is returning some sort of iterator here, and you're converting it to a list so that you can take the length of it below for use in constructing the progress information. I worry that this may impose significant overhead (both in memory consumption and latency) on scans of large repositories.

Is it possible to defer the list conversion and then perform it only in the case `self.git_options.progress` is true?

Reply:
I agree. Updated. Unfortunately I need to convert to list for the progressbar, otherwise it wont show progress.

Reply:
Also rebased and addressed merge conflicts.